lib/gmail/gmail.go

```diff
@@ -60,27 +63,29 @@ var (
 	ConcurrentDownloads = 8
 )
 
-// Gmail represents a Gmail client.
-type Gmail struct {
-	label    string
-	labelId  string
-	cache    gmailCache
-	svc      gmailService
-	dir      maildir.Maildir
-	progress chan<- lib.Progress
-}
-
-// Creates a new Gmail synchronizer.
-func NewGmail(dir string, label string) (*Gmail, error) {
-	g := Gmail{
-		label: label,
+// This function creates a JWT (JSON Web Token) HTTP client using a JSON
+// key file with the ability to impersonate any given gmail user of the
+// domain.
+func newJWTClient(serviceAccountJSONFile string, toImpersonate string) (*http.Client, error) {
+	// Read the JSON account file content.
+	data, err := ioutil.ReadFile(serviceAccountJSONFile)
+	if err != nil {
+		return nil, err
 	}
-	f := path.Join(dir, cacheFile)
-	if c, err := lib.NewBoltCache(f); err != nil {
+	// Create a JWT configuration object asking for Gmail read only access.
+	config, err := google.JWTConfigFromJSON(data, gmail.GmailReadonlyScope)
+	if err != nil {
 		return nil, err
-	} else {
-		g.cache = gmailCache{c}
 	}
+	// Add email to impersonate to the JWT configuration.
+	// Note: this is a NOP if toImpersonate is an empty string.
+	config.Subject = toImpersonate
+	// Create the http client and return it.
```

Comment:
Super nit: This comment isn't really necessary, since it's obvious we're creating the http client and returning it. :)